specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/dedicatedhsm.json

```diff
@@ -0,0 +1,606 @@
+              },
+              {
+                "value": "Provisioning",
+                "description": "The dedicated HSM is currently being provisioned."
+              },
+              {
+                "value": "Allocating",
+                "description": "A device is currently being allocated for the dedicated HSM resource."
+              },
+              {
+                "value": "Connecting",
+                "description": "The dedicated HSM is being connected to the virtual network."
+              },
+              {
+                "value": "Failed",
+                "description": "Provisioning of the dedicated HSM has failed."
+              },
+              {
+                "value": "CheckingQuota",
+                "description": "Validating the subscription has sufficient quota to allocate a dedicated HSM device."
+              },
+              {
+                "value": "Deleting",
+                "description": "The dedicated HSM is currently being deleted."
+              }
+            ]
+          }
+        }
+      },
+      "description": "Properties of the dedicated hsm"
+    },
+    "DedicatedHsm": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/DedicatedHsmProperties",
+          "description": "Properties of the dedicated HSM"
+        }
+      },
+      "allOf": [{
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "required": [
+        "location",
+        "sku",
+        "properties"
+      ],
+      "description": "Resource information with extended details."
+    },
+    "DedicatedHsmPatchParameters": {
+      "properties": {
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "description": "Patchable properties of the dedicated HSM"
+    },
+    "DedicatedHsmListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DedicatedHsm"
+          },
+          "description": "The list of dedicated HSMs."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of dedicated hsms."
+        }
+      },
+      "description": "List of dedicated HSMs"
+    },
+    "ResourceListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Resource"
+          },
+          "description": "The list of dedicated HSM resources."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of dedicated HSM resources."
+        }
+      },
+      "description": "List of dedicated HSM resources."
+    },
+    "KeyVaultError": {
```

Comment:
Is the usage of KeyVaultError intentional? Could it be say DedicatedHsmError

Reply:
Yes, please use DedicatedHsmError. It was my fault.